### PR TITLE
add applied manifestwork eviction test.

### PR DESCRIPTION
**Pull request by morvencao**

### CODE REVIEW

File: test/e2e/pkg/work_eviction_test.go [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+					}
+					if pod.Status.ContainerStatuses[0].State.Running == nil {
+						return fmt.Errorf("maestro server container not in running state")
+					}
+				}
+				return nil
+			}, 1*time.Minute, 1*time.Second).ShouldNot(HaveOccurred())
+		})
+
+		It("restart maestro agent again", func() {
```
Comment (clyang82) [inline test/e2e/pkg/work_eviction_test.go:138]: will not send the resync request automatically by the agent if timeout ocurrs?

Comment (morvencao) [reply test/e2e/pkg/work_eviction_test.go:138]: no, agent will not send resync request again.

**Review by clyang82 — APPROVED** [review]
LGTM

**Pull request merged**